frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.test.tsx

```diff
@@ -63,71 +57,129 @@ describe('AccessControlTab', () => {
+    screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_organisation'),
     });
-    expect(organisationCheckbox).not.toBeChecked();
-
-    const personCheckbox = screen.getByRole('checkbox', {
+    screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_person'),
     });
-    expect(personCheckbox).not.toBeChecked();
-
-    const subUnitCheckbox = screen.getByRole('checkbox', {
+    screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_sub_unit'),
     });
-    expect(subUnitCheckbox).not.toBeChecked();
   });
 
-  it('handles checkbox changes', async () => {
-    const user = userEvent.setup();
+  it('should render all checkboxes as checked when applicationMetadata contains all partyTypes allowed', async () => {
+    await resolveAndWaitForSpinnerToDisappear();
+    const checkboxes = screen.queryAllByRole('checkbox');
+    expect(checkboxes).toHaveLength(5);
+    checkboxes.forEach((c) => expect(c).toBeTruthy());
+  });
+
+  it('should render all checkboxes labels', async () => {
     await resolveAndWaitForSpinnerToDisappear();
 
-    const organisationCheckboxBefore = screen.getByRole('checkbox', {
+    screen.getByRole('row', {
+      name: textMock('settings_modal.access_control_tab_option_bankruptcy_estate'),
+    });
+    screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_organisation'),
     });
-    expect(organisationCheckboxBefore).not.toBeChecked();
+    screen.getByRole('row', {
+      name: textMock('settings_modal.access_control_tab_option_person'),
+    });
+    screen.getByRole('row', {
+      name: textMock('settings_modal.access_control_tab_option_sub_unit'),
+    });
+  });
 
-    await act(() => user.click(organisationCheckboxBefore));
+  it('should render the text of the button for help text correctly', async () => {
+    const user = userEvent.setup();
+    render();
+    const helpButton = screen.getByRole('button', {
+      name: textMock('settings_modal.access_control_tab_help_text_title'),
+    });
+    await act(() => user.click(helpButton));
+    screen.getByText(textMock('settings_modal.access_control_tab_help_text_heading'));
+    expect(
+      screen.getByText(textMock('settings_modal.access_control_tab_help_text_heading')),
+    ).toBeInTheDocument();
+  });
+
+  it('renders the documentation link with the correct text', async () => {
+    await resolveAndWaitForSpinnerToDisappear();
+    const documentationLink = screen.getByText(
+      textMock('settings_modal.access_control_tab_option_access_control_docs_link_text'),
+    );
+    expect(documentationLink).toBeInTheDocument();
+  });
 
+  it('render modal', async () => {
+    const user = userEvent.setup();
+    await resolveAndWaitForSpinnerToDisappear();
+    const checkbox = screen.getAllByRole('checkbox')[4];
+    await act(() => user.click(checkbox));
```

Comment:
Its better to get the actual checkbox you want to trigger and not getAll

Reply:
Fixed